src/helpers/prss.rs

```diff
@@ -218,7 +272,7 @@ impl GeneratorFactory {
 }
 
 /// The basic generator.  This generates values based on an arbitrary index.
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```

Comment:
wondering if clone causes secret key bytes to be copied as well and how bad it is if it is the case

Reply:
I looked into this and it's not ideal, but nor is it particularly bad.  The clone duplicates the expanded keys, which for most implementations is 11 `__mi128` variables.  That's large enough to be noticeable, but I don't think that it is a problem for what this would be used for (the permutation).